packages/wallet-sdk/src/connection/WalletLinkHTTP.ts

```diff
@@ -13,7 +13,7 @@ export class WalletLinkHTTP {
   }
 
   // mark unseen events as seen
-  private async markUnseenEventsAsSeen(events: ServerMessageEvent[]) {
+  private async markUnseenEventsAsSeen(events: ServerMessage<'Event'>[]) {
```

Comment:
and this new data structures can be applied nicely to the HTTP API client as well.
when calling `events` endpoint, it returns `ServerMessage<'Event'>`